markdown/core/src/main/kotlin/org/jetbrains/jewel/markdown/rendering/DefaultMarkdownBlockRenderer.kt

```diff
@@ -99,20 +92,22 @@ public open class DefaultMarkdownBlockRenderer(
             is BlockQuote -> render(block, rootStyling.blockQuote)
             is FencedCodeBlock -> render(block, rootStyling.code.fenced)
             is IndentedCodeBlock -> render(block, rootStyling.code.indented)
-            is H1 -> render(block, rootStyling.heading.h1)
-            is H2 -> render(block, rootStyling.heading.h2)
-            is H3 -> render(block, rootStyling.heading.h3)
-            is H4 -> render(block, rootStyling.heading.h4)
-            is H5 -> render(block, rootStyling.heading.h5)
-            is H6 -> render(block, rootStyling.heading.h6)
+            is Heading -> when (block.level) {
+                1 -> render(block, rootStyling.heading.h1)
+                2 -> render(block, rootStyling.heading.h2)
+                3 -> render(block, rootStyling.heading.h3)
+                4 -> render(block, rootStyling.heading.h4)
+                5 -> render(block, rootStyling.heading.h5)
+                6 -> render(block, rootStyling.heading.h6)
+                else -> error("$block")
```

Comment:
Would be nice to have a more meaningful error description here
```suggestion
                else -> error("Heading level ${block.level} not supported:\n$block")
```